src/hooks/useFetchPortfolios.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+import { useInfiniteQuery } from "react-query";
+import _ from "lodash";
+import { getPortfolioList } from "../apis/portfolio";
+
+export default function useFetchPortfolios() {
+  const [portfolios, setPortfolios] = useState([]);
+  const infiniteQuery = useInfiniteQuery(
+    ["portfolios"],
+    ({ pageParam = 1 }) => getPortfolioList(pageParam),
+    {
+      getNextPageParam: (lastPage, allPages) => {
+        if (lastPage.length < 10) {
+          return undefined;
+        }
+        return allPages.length;
+      },
+      keepPreviousData: true,
+    },
+  );
+
+  useEffect(() => {
+    if (infiniteQuery.data) {
+      const allFetchedPortfolios = infiniteQuery.data.pages.flat();
+      setPortfolios((prev) =>
+        _.unionBy([...prev, ...allFetchedPortfolios], "id"),
+      );
+    }
+  }, [infiniteQuery.data]);
+
+  return {
+    portfolios,
+    ...infiniteQuery,
+  };
```

Comment:
데이터 페칭한 결과물을 반환하기 위한 상태를 portfolios라고 만드셨는데
useInfiniteQuery에 의해 무한스크롤 결과물이 업데이트되면 InifiniteQuery도 변경이 될 겁니다
그러면 이 훅 내부 코드가 재실행될 것이고 이를 이용하면 useEffect를 사용하지 않을 수 있습니다.
그냥 변수를 선언하고 이를 return 시켜주기만 해도 InifiniteQuery가 변경될 때마다 업데이트되는 변수를 리턴해줄 수 있는거죠

```suggestion
  const allFetchedPortfolios = infiniteQuery.data.pages.flat();
  
  return {
    portfolios: allFetchedPortfolios,
    ...infiniteQuery,
  };
```

Reply:
멘토님께서 알려주신 cursor based 구현 방식으로 수정했습니다!!